### PR TITLE
Removed kubenet reference from docs

### PR DESCRIPTION
**Pull request by mk46**
Removing reference of `kubenet` from docs.
/kind cleanup
/language en
Fixes https://github.com/kubernetes/website/issues/31659

### CODE REVIEW

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 1]
```diff
@@ -15,7 +15,6 @@ Network plugins in Kubernetes come in a few flavors:
 
 * CNI plugins: adhere to the [Container Network Interface](https://github.com/containernetworking/cni) (CNI) specification, designed for interoperability.
   * Kubernetes follows the [v0.4.0](https://github.com/containernetworking/cni/blob/spec-v0.4.0/SPEC.md) release of the CNI specification.
-* Kubenet plugin: implements basic `cbr0` using the `bridge` and `host-local` CNI plugins
```
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md:18]: It sounds like the _only_ game in town is now CNI plugins. If that's true (I think it is), we need to rewrite the page overview to make that clear.

File: content/en/docs/tasks/network/validate-dual-stack.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ This document shares how to validate IPv4/IPv6 dual-stack enabled Kubernetes clu
 
 
 * Provider support for dual-stack networking (Cloud provider or otherwise must be able to provide Kubernetes nodes with routable IPv4/IPv6 network interfaces)
-* A [network plugin](/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/) that supports dual-stack (such as Calico, Cilium or Kubenet)
+* A [network plugin](/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/) that supports dual-stack (such as Calico or Cilium)
```
Comment (sftim) [inline content/en/docs/tasks/network/validate-dual-stack.md:19]: It's actually better not to list examples:
```suggestion
* A [network plugin](/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/)
  that supports dual-stack networking.
```

File: content/en/docs/setup/best-practices/node-conformance.md [hunk 3]
```diff
@@ -30,8 +30,6 @@ To run the node conformance test, perform the following steps:
     Because the test framework starts a local control plane to test the kubelet,
     use `http://localhost:8080` as the URL of the API server.
     There are some other kubelet command line parameters you may want to use:
-  * `--pod-cidr`: If you are using `kubenet`, you should specify an arbitrary CIDR
-    to Kubelet, for example `--pod-cidr=10.180.0.0/24`.
```
Comment (sftim) [inline content/en/docs/setup/best-practices/node-conformance.md:34]: I think you _might_ need to specify a Pod CIDR with some CNI plugins - can anyone confirm either way?

File: content/en/docs/concepts/services-networking/dual-stack.md [hunk 4]
```diff
@@ -43,7 +43,7 @@ The following prerequisites are needed in order to utilize IPv4/IPv6 dual-stack
      Kubernetes versions, refer to the documentation for that version
      of Kubernetes.
    * Provider support for dual-stack networking (Cloud provider or otherwise must be able to provide Kubernetes nodes with routable IPv4/IPv6 network interfaces)
-   * A network plugin that supports dual-stack (such as Kubenet or Calico)
+   * A network plugin that supports dual-stack (such as Calico)
```
Comment (sftim) [inline content/en/docs/concepts/services-networking/dual-stack.md:46]: It's actually better not to list examples:
```suggestion
   * A [network plugin](/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/)
    that supports dual-stack networking.
```

**Review by sftim — CHANGES_REQUESTED** [review]
Thanks @mk46 

I have some inline feedback - I hope it's useful.

Also one piece to suggest a change you _haven't_ made. For the dockershim removal, we should also edit:
```markdown
The kubelet has a single default network plugin, and a default network common to the entire cluster. It probes for plugins when it starts up, remembers what it finds, and executes the selected plugin at appropriate times in the pod lifecycle (this is only true for Docker, as CRI manages its own CNI plugins). There are two Kubelet command line parameters to keep in mind when using plugins:
```

to avoid mentioning Docker [Engine].

**Comment by tengqm** [comment]
@sftim I'd vote for one more revision before kicking this in.

**Comment by reylejano** [comment]
Fixes issue https://github.com/kubernetes/website/issues/31659

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md (hunk 1 above)
Comment (mk46) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md:18]: Thanks @sftim, In https://github.com/kubernetes/kubernetes/pull/106907, `--network-plugin` was removed. Can we remove this file?

Comment (sftim) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md:18]: Which file do you propose to remove @mk46?

Comment (mk46) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md:18]: `content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md`

**Comment by tengqm** [comment]
@mk46 Would you like give this PR another stab?

**Comment by mk46** [comment]
Hi @tengqm,  I would like to remove the below pages in favor of https://github.com/kubernetes/kubernetes/blob/release-1.23/CHANGELOG/CHANGELOG-1.23.md#ipv4ipv6-dual-stack-networking-graduates-to-ga

- `content/en/docs/tasks/network/validate-dual-stack.md`
- `content/en/docs/concepts/services-networking/dual-stack.md`
and also  `content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md` in favor of removal of `--network-plugin` in https://github.com/kubernetes/kubernetes/pull/106907

Are these changes make sense?

**Comment by tengqm** [comment]
> Hi @tengqm, I would like to remove the below pages in favor of https://github.com/kubernetes/kubernetes/blob/release-1.23/CHANGELOG/CHANGELOG-1.23.md#ipv4ipv6-dual-stack-networking-graduates-to-ga
> 
> * `content/en/docs/tasks/network/validate-dual-stack.md`
> * `content/en/docs/concepts/services-networking/dual-stack.md`
>   and also  `content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md` in favor of removal of `--network-plugin` in [Clean up dockershim flags in the kubelet kubernetes#106907](https://github.com/kubernetes/kubernetes/pull/106907)
> 
> Are these changes make sense?

Maybe you can help file an issue about the problem to address. My understanding of this PR is that it is about removal of kubenet.

**Comment by sftim** [comment]
@mk46, we still need the following pages:

- https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/
- https://kubernetes.io/docs/tasks/network/validate-dual-stack/

I'm not so sure about https://kubernetes.io/docs/concepts/services-networking/dual-stack/ but we should not simply _remove_ it. We might possibly want to reshape it to discuss the three network family options (single stack IPv6, single stack IPv4, dual stack) on an equal footing. Alongside such a change we could add a page about validating single-stack networking, and maybe an overview page about validating that your cluster is healthy.

For v1.24 we need a smaller change that merely omits mention of kubenet. The changes you've proposed are about right; my previous feedback still applies though.

**Comment by tengqm** [comment]
With respect to changes to dual-stack contents, I agree to @sftim's suggestions.

File: content/en/docs/tasks/network/validate-dual-stack.md (hunk 2 above)
Comment (mk46) [reply content/en/docs/tasks/network/validate-dual-stack.md]: done.

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 5]
```diff
@@ -11,17 +11,16 @@ weight: 10
 
 <!-- overview -->
 
-Network plugins in Kubernetes come in a few flavors:
+Network plugins in Kubernetes come only as CNI plugins:
 
 * CNI plugins: adhere to the [Container Network Interface](https://github.com/containernetworking/cni) (CNI) specification, designed for interoperability.
   * Kubernetes follows the [v0.4.0](https://github.com/containernetworking/cni/blob/spec-v0.4.0/SPEC.md) release of the CNI specification.
```
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md:17]: A possible further tweak:
```suggestion
Kubernetes {{< skew currentVersion >}} supports [Container Network Interface](https://github.com/containernetworking/cni)
(CNI) plugins for cluster networking. You must use a CNI plugin that is compatible with your cluster
and that suits your needs. Different plugins are available (both open- and closed- source) in the
wider Kubernetes ecosystem.

You must use a CNI plugin that is compatible with the
[v0.4.0](https://github.com/containernetworking/cni/blob/spec-v0.4.0/SPEC.md) release of the CNI specification.
```

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 6]
```diff
@@ -11,17 +11,16 @@ weight: 10
...
-* Kubenet plugin: implements basic `cbr0` using the `bridge` and `host-local` CNI plugins
 
 <!-- body -->
 
 ## Installation
 
-The kubelet has a single default network plugin, and a default network common to the entire cluster. It probes for plugins when it starts up, remembers what it finds, and executes the selected plugin at appropriate times in the pod lifecycle (this is only true for Docker, as CRI manages its own CNI plugins). There are two Kubelet command line parameters to keep in mind when using plugins:
+The kubelet has a single default network plugin, and a default network common to the entire cluster.The CRI manages its own CNI plugins. There are two Kubelet command line parameters to keep in mind when using plugins:
```
Comment (sftim) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md:23]: This revised paragraph might need technical review. I don't think there is a default network plugin for the kubelet now that `dockershim` is gone.

File: content/en/docs/concepts/services-networking/dual-stack.md (hunk 4 above)
Comment (mk46) [reply content/en/docs/concepts/services-networking/dual-stack.md]: Done

**Review by sftim — COMMENTED** [review]
A bit more feedback

**Review by sftim — COMMENTED** [review]
I think this might be barely OK to merge, provided that we go back and do more fixups.

**Comment by sftim** [comment]
/lgtm
for SIG Docs

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md (hunk 5 above)
Comment (mk46) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md]: done.

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md [hunk 7]
```diff
@@ -11,17 +11,17 @@ weight: 10
...
-* Kubenet plugin: implements basic `cbr0` using the `bridge` and `host-local` CNI plugins
+You must use a CNI plugin that is compatible with the
+[v0.4.0](https://github.com/containernetworking/cni/blob/spec-v0.4.0/SPEC.md) release of the CNI specification.
 
 <!-- body -->
 
 ## Installation
 
-The kubelet has a single default network plugin, and a default network common to the entire cluster. It probes for plugins when it starts up, remembers what it finds, and executes the selected plugin at appropriate times in the pod lifecycle (this is only true for Docker, as CRI manages its own CNI plugins). There are two Kubelet command line parameters to keep in mind when using plugins:
+The kubelet has a single default network plugin, and a default network common to the entire cluster.The CRI manages its own CNI plugins. There are two Kubelet command line parameters to keep in mind when using plugins:
```
Comment (reylejano) [inline content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md:24]:
```suggestion
The kubelet has a single default network plugin, and a default network common to the entire cluster. The CRI manages its own CNI plugins. There are two Kubelet command line parameters to keep in mind when using plugins:
```
nit: missing space, can be fixed in a follow-up PR

**Comment by sftim** [comment]
I'm happy to see this merged, I do hope we also get the nits tidied up.
/approve

**Comment by sftim** [comment]
@mk46 would you be willing to open a new PR to further tweak that page based on the feedback in _this_ PR?

File: content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md (hunk 7 above)
Comment (sftim) [reply content/en/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md:24]: > The kubelet has a single default network plugin

Is this true for v1.24 and later?

**Comment by mk46** [comment]
> @mk46 would you be willing to open a new PR to further tweak that page based on the feedback in _this_ PR?

Done https://github.com/kubernetes/website/pull/31849

**Pull request merged**